### PR TITLE
Prepare Synapse libraries for release

### PR DESCRIPTION
**Pull request by annelo-msft**
Update the Synapse C# libraries to support new service API versions:

- Update `Azure.Analytics.Synapse.AccessControl` from `2020-08-01-preview` to `2020-12-01`
- Update `Azure.Analytics.Synapse.Spark` from `2019-11-01-preview` to `2020-12-01`

In addition, add support for multiple service versions to `Azure.Analytics.Synapse.ManagedPrivateEndpoints`.  This library will now support both `2020-12-01` and `2021-06-01-preview` versions of the service.

### CODE REVIEW

File: sdk/synapse/Azure.Analytics.Synapse.Spark/src/autorest.md [hunk 1]
```diff
@@ -22,4 +22,12 @@ directive:
...
-```
\ No newline at end of file
+```
+
+``` yaml
+directive:
+- from: swagger-document
+  where: $.parameters.LivyApiVersion
+  transform: >
+    $["x-ms-client-default"] = $.default;
```
Comment (christothes) [inline sdk/synapse/Azure.Analytics.Synapse.Spark/src/autorest.md:32]: jus curious why this is needed

**Review by christothes — APPROVED** [review]

File: sdk/synapse/Azure.Analytics.Synapse.Spark/src/autorest.md (hunk 1 above)
Comment (annelo-msft) [reply sdk/synapse/Azure.Analytics.Synapse.Spark/src/autorest.md:32]: It is needed to ensure the constructor sets a default parameter for `livyApiVersion` -- apparently the AutoRest behavior recently changed.

From @jianghaolu:

AutoRest recently changed the behavior for `default` to not send the value and rely on the service to "choose the default". `x-ms-client-default` does what default used to do.

**Comment by annelo-msft** [comment]
Due to:
- an [issue in the C# Code Generator](https://github.com/Azure/autorest.csharp/issues/1454) blocking generation of the Artifacts
- an issue in the Synapse Monitoring service where `api-version 2020-12-01` is unsupported

We will not release the C# Synapse libraries this month.  The next release is planned for September.

**Comment by annelo-msft** [comment]
We will not release Artifacts or Monitoring this month, but will release the remaining three libraries.

**Review by christothes — APPROVED** [review]

File: sdk/synapse/Azure.Analytics.Synapse.Spark/src/autorest.md (hunk 1 above)
Comment (annelo-msft) [reply sdk/synapse/Azure.Analytics.Synapse.Spark/src/autorest.md]: This was removed with an update to the swagger file directly.

**Pull request merged**